xenmgr/Vm/Utility.hs

```diff
@@ -60,6 +60,18 @@ mount dev dir ro = void $ readProcessOrDie "mount" ["-o", opts, dev, dir] "" whe
   ro_opt | ro = "ro"
          | otherwise = ""
 
+mount_try_n n dev dir ro =
+  do
+    info $ "mount " ++ (show n) ++ " " ++ dev ++ " on " ++ (show dir) ++ " read-only:" ++ (show ro)
```

Comment:
I would add a 
```haskell
threadDelay (10^6)
```
just to give it some time.  I bet there's some inherent delay waiting for udevadm settle to return, but there's nothing wrong with pausing explicitly for a second on each retry loop.